model/model.py

```diff
@@ -232,3 +234,785 @@ def __str__(self):
 
     def __repr__(self):
         return f"RedshiftInception({self.input_shape[1:]},{self.output_shape[1]})"
+
+
+class RedshiftClassifierInceptionV4(Model):
+    """An inception model version 4.
+
+    This model is uses a combination of prescriptions from the
+    redshift paper (http://arxiv.org/abs/1806.06607) as well as
+    IncpetionV4 paper (https://arxiv.org/abs/1602.072).At its current
+    form the Inception network uses hyper-parameters used by classification
+    (input shape (299, 299, 3)), with PReLU activation for the convolution
+    layers and all the MaxPooling layers replaced by AveragePooling.
+
+    ToDo: Determining correct set of hyper-parameters for RedShift Classification
+
+    Parameters
+    ----------
+    input_image_shape : tuple
+        The shape of the input Image
+    keep_probability : float, default=0.8
+        The keep probability for the dropout layer
+    num_redshift_bins : int
+        The number of redshift classes for the end softmax layer
+
+    Notes
+    -----
+    This Network assumes that the data format is `channels_last`
+    """
+    def __init__(self,
+                 input_image_shape=(299, 299, 3),
```

Comment:
I suggest changing the default value of `input_image_shape` to `(64,64,5)` to reflect the actual expected shape of photometric images.

Reply:
At this current form, An input image size of (64, 64, 5) is incompatible with the `stem-block` used here. This is because the stem block reduces the image size by almost 8 folds from (299, 299, 3) to (35 * 35 * 384). For a 64 * 64 * 5 image we would end up with ( 8 * 8 * 384). This is not logical to use as it would endup not being compatible with the `Reduction-A` and `Reduction-B` blocks down in the architecture.

Reply:
That makes sense. My source of confusion right now is how would this be compatible with the images we are using? Will we need an additional dense layer between the input and the call to `add_stem` that reshapes the `(64,64,5)` image to the expected `(299,299,3)`?

Reply:
We should do some research on how to replace this stem block. The motivation behind using a these blocks is learning equivalent features with less parameters. In the origninal redshift paper, they have used a 5*5 kernel with an Average pooling block before using inception blocks. Not sure how to fair this one out.

Reply:
Basically, if you play with the parameters like `strides` and `num_filters` you can get an appropriate representation to feed forward to the inception blocks.

Reply:
https://www.kaggle.com/babbler/inception-model-for-fashion-mnist/notebook

This notebook has some interesting usage of inceptionV4 modules.